src/Build/BackEnd/Components/Logging/LoggingServiceLogMethods.cs

```diff
@@ -420,6 +420,18 @@ public void LogBuildFinished(bool success)
             WaitForLoggingToProcessEvents();
         }
 
+        /// <inheritdoc />
+        public void LogBuildCanceled()
+        {
+            string message = ResourceUtilities.GetResourceString("AbortingBuild"); 
+            BuildCanceledEventArgs buildEvent = new BuildCanceledEventArgs(message);
+
+            ProcessLoggingEvent(buildEvent);
+
+            // Make sure we process this event before going any further
+            WaitForLoggingToProcessEvents();
```

Comment:
Is this required for cancellation? Does it provide meaningful guarantees across processes? If not I might expect the exact timing of the cancellation to be advisory, and maybe we don't need to block here?

Reply:
I think you are right. We don't need this for cancellation.